fetching/tarball-fetcher/src/remoteTarballFetcher.ts

```diff
@@ -142,26 +143,26 @@ export function createDownloader (
           })
         }
 
-        const data: Buffer = Buffer.from(new SharedArrayBuffer(downloaded))
+        data = Buffer.from(new SharedArrayBuffer(downloaded))
         let offset: number = 0
         for (const chunk of chunks) {
           chunk.copy(data, offset)
           offset += chunk.length
         }
-        return await addFilesFromTarball({
-          buffer: data,
-          cafsDir: opts.cafs.cafsDir,
-          readManifest: opts.readManifest,
-          integrity: opts.integrity,
-          filesIndexFile: opts.filesIndexFile,
-          url,
-          pkg: opts.pkg,
-        })
       } catch (err: any) { // eslint-disable-line
         err.attempts = currentAttempt
         err.resource = url
         throw err
       }
+      return addFilesFromTarball({
```

Comment:
Nice. I like the move to put this outside of the `try` block.

Reply:
Right, I don't know how it got into the try/catch but it shouldn't have been there. It resulted in fetch retries when the correctly downloaded tarball couldn't be unpacked and added to the store.